client/signup/steps/import-url/index.jsx

```diff
@@ -80,23 +61,95 @@ class ImportURLStepComponent extends Component {
 
 	focusInput = () => invoke( this.inputRef, 'focus' );
 
+	setUrlError = urlValidationMessage =>
+		this.setState(
+			{
+				urlValidationMessage,
+			},
+			this.focusInput
+		);
+
 	handleSubmit = event => {
 		event.preventDefault();
 		const isValid = this.validateUrl();
 
 		if ( ! isValid ) {
-			this.focusInput();
 			return;
 		}
 
-		const { urlInputValue, stepName } = this.props;
+		const { stepName, translate, urlInputValue } = this.props;
 
-		this.props
-			.submitImportUrlStep( {
-				siteUrl: urlInputValue,
-				stepName,
-			} )
-			.then( this.props.goToNextStep );
+		this.setState( {
+			isLoading: true,
+			urlValidationMessage: '',
+		} );
+
+		wpcom
+			.undocumented()
+			.isSiteImportable( urlInputValue )
+			.then(
+				( {
+					site_engine: siteEngine,
+					site_favicon: siteFavicon,
+					site_status: siteStatus,
+					site_title: siteTitle,
+					site_url: siteUrl,
+					importer_types: importerTypes,
+				} ) => {
+					if ( 404 === siteStatus ) {
+						return this.setUrlError(
+							translate( 'The URL entered was not found. Double-check the URL and try again.' )
+						);
+					}
+
+					if ( includes( importerTypes, 'url' ) && 200 !== siteStatus ) {
+						return this.setUrlError(
+							translate( "We're not able to reach that URL. Double-check the URL and try again." )
+						);
+					}
+
+					if ( 'unknown' !== siteEngine && isEmpty( importerTypes ) ) {
```

Comment:
These conditions mean we're sure we don't have an importer for the site. The best case here seems like the user continues with the normal onboarding flow to create a site. I think you're right that an error isn't really the best way to show this--we need an alternative design for this case.